indra/newview/llviewermenu.cpp

```diff
@@ -6505,7 +6506,19 @@ class LLAvatarResetSkeleton: public view_listener_t
         }
 		if(avatar)
         {
-            avatar->resetSkeleton(false);
+			bool owned = false;
+			if(avatar->isAnimatedObject())
+			{
+				owned = avatar->mOwnerID == gAgent.getID();
+			}
+			else
+			{
+				owned = avatar->getID() == gAgent.getID();
+			}
+			LLHUDEffectResetSkeleton* effectp = (LLHUDEffectResetSkeleton*)LLHUDManager::getInstance()->createViewerEffect(LLHUDObject::LL_HUD_EFFECT_RESET_SKELETON, owned);
+			effectp->setSourceObject(gAgentAvatarp);
+			effectp->setTargetObject((LLViewerObject*)avatar);
+			effectp->setResetAnimations(false);
```

Comment:
If the point is to allow resetting selected objects, you should check not just ownership but LLPermissions.